GeoInterfaceMakie/src/GeoInterfaceMakie.jl

```diff
@@ -23,12 +23,32 @@ function _convert_arguments(t, geom)::Tuple
     geob = GI.convert(GB, geom)
     MC.convert_arguments(t, geob)
 end
+
 function _convert_array_arguments(t, geoms::AbstractArray{T})::Tuple where T
     if Missing <: T
         geob = map(geom -> GI.convert(GB, geom), skipmissing(geoms))
     else
         geob = map(geom -> GI.convert(GB, geom), geoms)
     end
+    if !(eltype(geob) <: GB.AbstractGeometry) || eltype(geob) isa Union # Unions are bad
+        if isempty(geob)
+            geob = geob
+        end
+        first_trait = GI.geomtrait(first(geob))
+        last_trait = GI.geomtrait(last(geob))
```

Comment:
What if the first and last are the same but the middle is different?

Reply:
We already know that something is different because the types are not the same.  But a more robust check would be nice.